utils/build/build.js

```diff
@@ -67,71 +70,53 @@ function main() {
 		}
 
 	}
-	
+
 	if ( args.amd ){
 		buffer.push('exports.THREE = THREE;\n\n} ) );');
 	};
-	
+
 	var temp = buffer.join( '' );
-	
+
 	if ( !args.minify ){
 
 		fs.writeFileSync( output, temp, 'utf8' );
+		console.log( '  Compile Time: ' + (Date.now() - startTime)/1000 + 's');
 
 	} else {
 
+		var tempSourceFilename = random();
+		fs.writeFileSync( tempSourceFilename, temp, 'utf8' );
+
 		var LICENSE = "threejs.org/license";
 
 		// Parsing
 
-		var toplevel = null;
-
-		toplevel = uglify.parse( '// ' + LICENSE + '\n' );
-
-		sources.forEach( function( source ) {
-
-			toplevel = uglify.parse( source.contents, {
-				filename: source.file,
-				toplevel: toplevel
-			} );
-
-		} );
-
-		// Compression
-
-		toplevel.figure_out_scope();
-		var compressor = uglify.Compressor( {} );
-		var compressed_ast = toplevel.transform( compressor );
-
-		// Mangling
-
-		compressed_ast.figure_out_scope();
-		compressed_ast.compute_char_frequency();
-		compressed_ast.mangle_names();
-
-		// Output
-
-		var source_map_options = {
-			file: 'three.min.js',
-			root: 'src'
+		var closureOptions = {
+		  js: tempSourceFilename,
+      externs: args.externs,
+			jscomp_off: [ 'checkTypes', 'globalThis' ],
+      compilation_level: 'SIMPLE',
```

Comment:
No, his whitespace is fine, I guess. That file contained quirky spaces and I removed them too - we'll have fun merging :).